hack/create.sh

```diff
@@ -16,7 +16,10 @@ USE_PROD_FLAG="-use-prod=false"
 if [[ -n "$TEST_IN_PRODUCTION" ]]; then
     USE_PROD_FLAG="-use-prod=true"
 else
-	hack/fakerp.sh $RESOURCEGROUP &
+    go generate ./...
+    go run cmd/fakerp/main.go &
 fi
 
+trap 'return_id=$?; kill $(lsof -t -i :8080); exit $return_id' EXIT
```

Comment:
do `return_id=$?;  kill $(lsof -t -i :8080); wait $(lsof -t -i :8080) 2>/dev/null; exit $return_id` instead to suppress the `signal terminated` message

Reply:
What's the point of the `wait` command?

Reply:
without wait, `signal terminated` is returned in the bash terminal 

```
charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ go run cmd/fakerp/main.go &
++ generate_command_executed_sequence
++ printf '\e\7'
[1] 34951
+ go run cmd/fakerp/main.go
charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ INFO[2018-12-18T13:29:45+01:00] starting the fake resource provider          
INFO[2018-12-18T13:29:45+01:00] starting server on localhost:8080            
^C
charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ kill $(lsof -t -i :8080)
++ generate_command_executed_sequence
++ printf '\e\7'
++ lsof -t -i :8080
+ kill 35193
charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ signal: terminated
^C
[1]+  Exit 1                  go run cmd/fakerp/main.go
```

with wait

```
charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ go run cmd/fakerp/main.go &
++ generate_command_executed_sequence
++ printf '\e\7'
[1] 35360
+ go run cmd/fakerp/main.go
charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ INFO[2018-12-18T13:32:39+01:00] starting the fake resource provider          
INFO[2018-12-18T13:32:39+01:00] starting server on localhost:8080            
^C
charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ kill $(lsof -t -i :8080); wait $(lsof -t -i :8080)
++ generate_command_executed_sequence
++ printf '\e\7'
++ lsof -t -i :8080
+ kill 35593
++ generate_command_executed_sequence
++ printf '\e\7'
++ lsof -t -i :8080
signal: terminated
+ wait
charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ 
```

Reply:
fwiw, i used a debian system to test the changes. on fedora your changes work fine.

Reply:
more specifically, the wait command will wait for the given subprocess id to finish or, if no PID specified, will wait for all subprocesses spawned from the current shell to finish, sample:

```bash
#!/bin/bash

for x in {1..10};
do
  bash -c "sleep $x && echo \"Done waiting $x seconds\"" &
done

echo "Waiting..."
wait
echo "Done."
```

Reply:
Added `wait`. FWIW, the message is still not supressed so I removed the stderr redirection to /dev/null.